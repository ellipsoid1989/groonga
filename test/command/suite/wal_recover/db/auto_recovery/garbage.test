# _database_close/_database_reopen can't use with HTTP because requests via HTTP are always
# processed by workers. Workers never use primary WAL role.
#@require-interface stdio

#@require-feature message_pack

#$GROONGA_ALLOW_DATABASE_REOPEN=yes
#$GRN_WAL_ROLE=primary

table_create #recovering#Data TABLE_PAT_KEY ShortText
column_create #recovering#Data content COLUMN_SCALAR ShortText

table_create Data TABLE_PAT_KEY ShortText
column_create Data content COLUMN_SCALAR ShortText

dump

_database_close

#@copy-path #{db_path} #{db_path}.wal
#@add-important-log-levels notice
_database_reopen
#@remove-important-log-levels notice

dump
