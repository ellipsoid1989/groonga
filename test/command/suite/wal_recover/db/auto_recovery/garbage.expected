table_create #recovering#Data TABLE_PAT_KEY ShortText
[[0,0.0,0.0],true]
column_create #recovering#Data content COLUMN_SCALAR ShortText
[[0,0.0,0.0],true]
table_create Data TABLE_PAT_KEY ShortText
[[0,0.0,0.0],true]
column_create Data content COLUMN_SCALAR ShortText
[[0,0.0,0.0],true]
dump
table_create #recovering#Data TABLE_PAT_KEY ShortText
column_create #recovering#Data content COLUMN_SCALAR ShortText

table_create Data TABLE_PAT_KEY ShortText
column_create Data content COLUMN_SCALAR ShortText
_database_close
[[0,0.0,0.0],true]
_database_reopen
[[0,0.0,0.0],true]
#|e| [reader][read-entry][dat][recover][(temporary)] must be map: <positive-integer>(2): path:<db/db>
#|n| [db][wal][recover] rebuild broken DB keys
#|n| [db][wal][recover] succeeded to rebuild broken DB keys
#|n| DDL:256:obj_remove #recovering#Data
#|n| DDL:257:obj_remove #recovering#Data.content
#|n| spec:257:remove:#recovering#Data.content:65(column:var_size)
#|n| spec:256:remove:#recovering#Data:49(table:pat_key)
dump
table_create Data TABLE_PAT_KEY ShortText
column_create Data content COLUMN_SCALAR ShortText
